src/driver/swim.c

```diff
@@ -318,26 +317,34 @@ int swim_entry() {
   // 128 cycles (@8MHz) i.e. 16us. We currently just use it as a sanity
   // check to make sure the (right) device responds.
   timeout = MICROS_TO_CYCLES(30) / 6;
-  while (!READ_PIN() && --timeout)
+  while (!READ_PIN(SWIM) && --timeout)
     ;  // ~6 cycles
   int duration = get_ccount() - counter;
   if (!timeout) return SWIM_ERROR_SYNC_TIMEOUT_2;
 
   esp8266_leave_critical(state);
-  sync_ccount(counter + duration +
-              24);  // Need to wait at least 300ns (=24 cycles)
+
+  SET_PIN_HIGH(SWIM);
+  PIN_AS_INPUT(SWIM);
```

Comment:
I think you can remove these two lines. The pin should already be HIGH and in INPUT state (set ~20 lines above).

Reply:
At least looks like this in the aggregated diff: https://github.com/rumpeltux/esp-stlink/pull/6/files

Reply:
Removed